internal/shell_executable/shell_executable.go

```diff
@@ -141,6 +141,21 @@ func (b *ShellExecutable) WaitForTermination() (hasTerminated bool, exitCode int
 	}
 }
 
+func (b *ShellExecutable) ExitCode() int {
+	if b.cmd.ProcessState == nil {
+		// We are okay with waiting here,
+		// Since if the user's shell is still running, read would have timed out
```

Comment:
@ryan-gang this is a lot of assumptions for this level - we're in an "Executable" class, it has no idea of whether a read timed out or not, or why it's being called. It's also a bit unintuitive that `ExitCode` actually ends up waiting for a process to complete (it looks like a "getter"). 

Can we just expose `WaitForTermination` instead and proxy to `b.WaitForTermination`? That seems like there'd be less assumptions. It should be okay to call `WaitForTermination` twice, if I'm not wrong (worth confirming) - so we don't even need the `if b.cmd.ProcessState == nil` check.